ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py

```diff
@@ -3671,3 +3671,50 @@ def test_numpy_ndarray_view(
         frontend_method_data=frontend_method_data,
         on_device=on_device,
     )
+
+
+# trace
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="trace",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=["int64"],
```

Comment:
Hi @Nwaamaka-Iduwe 

The data generation for the test seems a bit restrictive to me. The `dtype` of the array should be `valid` or `numeric` since `np.trace` **can** accept floating point arrays as well so not sure why you've only limited it to `int64` 😅

Reply:
Hi, the problem here is that other tests keep failing if I don't restrict it. Only the numpy tests passes using only "numeric" and "valid". I've tried to use lambda to solve this and indexing but none of them work. The best I can get is an 80% pass on tests. If you have any suggestions, that'll be appreciated.

Reply:
Hi, the problem here is that other tests keep failing if I don't restrict it. Only the numpy tests passes using only "numeric" and "valid". I've tried to use lambda to solve this and indexing but none of them work. The best I can get is an 80% pass on tests. If you have any suggestions, that'll be appreciated.

Reply:
Do you know the reason why they were failing though? I just looked at the failing tests in a previous commit and the error seems to be a `dtype-mismatch` problem: 
```python
E   AssertionError: returned dtype = int64, ground-truth returned dtype = uint64
E   Falsifying example: test_numpy_trace(
E       backend_fw='jax',
E       frontend='numpy',
E       on_device='cpu',
E       dtype_x_axis=(['uint16'], [array([[0]], dtype=uint16)], (0, 1)),
E       offset=0,
```
Seems like the frontend function is returning an `int64` whereas the original `np.trace` is returning `uint64`. Not sure why `ivy.trace` is returning an incorrect dtype though🤔. Could you try debugging this further to figure out the exact cause?

Reply:
Hi @YushaArif99, this is what I was able to get from debugging the code: 

WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1695749278.850075   35972 tfrt_cpu_pjrt_client.cc:349] TfrtCpuClient created.
No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
/workspaces/ivy/ivy/utils/exceptions.py:390: UserWarning: The current backend: 'jax' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
  warnings.warn(
/workspaces/ivy/ivy/utils/exceptions.py:390: UserWarning: The current backend: 'tensorflow' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
  warnings.warn(
/workspaces/ivy/ivy/utils/exceptions.py:390: UserWarning: The current backend: 'paddle' does not support inplace updates natively. Ivy would quietly create new arrays when using inplace updates with this backend, leading to memory overhead (same applies for views). If you want to control your memory management, consider doing ivy.set_inplace_mode('strict') which should raise an error whenever an inplace update is attempted with this backend.
  warnings.warn(
[2023-09-26 17:28:12,182] [ WARNING] data_type.py:159 - This function includes the mixed partial function 'ivy.get_item'. Please note that the returned data types may not be exhaustive. Please check the dtypes of `ivy.get_item` for more details
[2023-09-26 17:28:12,210] [ WARNING] data_type.py:159 - This function includes the mixed partial function 'ivy.get_item'. Please note that the returned data types may not be exhaustive. Please check the dtypes of `ivy.get_item` for more details
[2023-09-26 17:28:12,226] [ WARNING] data_type.py:159 - This function includes the mixed partial function 'ivy.get_item'. Please note that the returned data types may not be exhaustive. Please check the dtypes of `ivy.get_item` for more details
[2023-09-26 17:28:12,577] [ WARNING] data_type.py:159 - This function includes the mixed partial function 'ivy.flatten'. Please note that the returned data types may not be exhaustive. Please check the dtypes of `ivy.flatten` for more details